### PR TITLE
Simple Refactor of System RawOrigin

### PR DESCRIPTION
**Pull request by shawntabrizi**
This refactors the `frame_system::RawOrigin` to now be part of `frame_support::dispatch`, and reduces some duplicate code.

Also exposes `ensure_signed_or_root` into pallet prelude.

Originally written by @gavofyork in https://github.com/paritytech/substrate/pull/10195/

### CODE REVIEW

**Review by KiChjang — APPROVED** [review]

File: frame/system/src/lib.rs [hunk 1]
```diff
@@ -1645,7 +1624,7 @@ impl<T: Config> Lookup for ChainContext<T> {
 
 /// Prelude to be used alongside pallet macro, for ease of use.
 pub mod pallet_prelude {
-	pub use crate::{ensure_none, ensure_root, ensure_signed};
+	pub use crate::{ensure_none, ensure_root, ensure_signed, ensure_signed_or_root};
```
Comment (bkchr) [inline frame/system/src/lib.rs:1627]: Why are these functions also not moved?

File: frame/support/src/dispatch.rs [hunk 2]
```diff
@@ -60,6 +61,28 @@ pub trait Callable<T> {
 // https://github.com/rust-lang/rust/issues/51331
 pub type CallableCallFor<A, R> = <A as Callable<R>>::Call;
 
+/// Origin for the System pallet.
```
Comment (bkchr) [inline frame/support/src/dispatch.rs:64]:
```suggestion
/// Inherent origin of FRAME
```

Or probably something better.

File: frame/support/src/dispatch.rs [hunk 3]
```diff
@@ -60,6 +61,28 @@ pub trait Callable<T> {
...
+#[derive(PartialEq, Eq, Clone, RuntimeDebug, Encode, Decode, TypeInfo)]
+pub enum RawOrigin<AccountId> {
```
Comment (bkchr) [inline frame/support/src/dispatch.rs:66]: And we should rename this imo.

File: frame/support/src/origin.rs [hunk 4]
```diff
@@ -0,0 +1,180 @@
+use crate::{
```
Comment (bkchr) [inline frame/support/src/origin.rs:1]: Missing license header

File: frame/support/src/origin.rs [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+	error::BadOrigin,
+	traits::{EnsureOrigin, SortedMembers},
+	RuntimeDebug,
+};
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+/// Origin for the System pallet.
```
Comment (bkchr) [inline frame/support/src/origin.rs:9]: Stale docs :P

File: frame/support/src/origin.rs [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+where
+	OuterOrigin: Into<Result<BaseOrigin<AccountId>, OuterOrigin>>,
+{
+	match o.into() {
+		Ok(BaseOrigin::None) => Ok(()),
+		_ => Err(BadOrigin),
+	}
+}
+
+pub struct EnsureRoot<AccountId>(sp_std::marker::PhantomData<AccountId>);
```
Comment (bkchr) [inline frame/support/src/origin.rs:81]: Missing docs here and below.

**Pull request closed without merging**